File: ios/ScreenMotion/Models/ThemeType.swift
import Foundation

enum ThemeType: String, CaseIterable, Identifiable, Codable {
    case spaceship
    case space
    case aquarium
    case car
    case truck
    case bus
    case tractor
    case tank
    case helicopter
    case airplane
    case bicycle
    case nature

    var id: String { rawValue }

    var title: String {
        switch self {
        case .spaceship: "Uzay Aracı"
        case .space: "Uzay"
        case .aquarium: "Akvaryum"
        case .car: "Araba"
        case .truck: "Kamyon"
        case .bus: "Otobüs"
        case .tractor: "Traktör"
        case .tank: "Tank"
        case .helicopter: "Helikopter"
        case .airplane: "Uçak"
        case .bicycle: "Bisiklet"
        case .nature: "Doğa"
        }
    }

    var emoji: String {
        switch self {
        case .spaceship: "🚀"
        case .space: "🌌"
        case .aquarium: "🐠"
        case .car: "🏎️"
        case .truck: "🚛"
        case .bus: "🚌"
        case .tractor: "🚜"
        case .tank: "🛡️"
        case .helicopter: "🚁"
        case .airplane: "✈️"
        case .bicycle: "🚲"
        case .nature: "🏔️"
        }
    }

    var assetName: String {
        switch self {
        case .spaceship, .space, .airplane: "space"
        case .aquarium: "aquarium"
        case .car, .truck, .bus: "car_night"
        case .tractor, .tank, .helicopter, .bicycle, .nature: "nature"
        }
    }

    var isVehicle: Bool {
        ![ThemeType.space, .aquarium, .nature].contains(self)
    }
}

File: android/app/src/main/java/com/screenmotion/app/model/ThemeType.kt
package com.screenmotion.app.model

enum class ThemeType(
    val title: String,
    val assetPath: String?,
    val emoji: String
) {
    SPACE("Uzay", "wallpapers/space.jpg", "🚀"),
    AQUARIUM("Akvaryum", "wallpapers/aquarium.jpg", "🐠"),
    CAR("Araba", "wallpapers/car_night.jpg", "🏎️"),
    TRUCK("Kamyon", "wallpapers/car_night.jpg", "🚛"),
    BUS("Otobüs", "wallpapers/car_night.jpg", "🚌"),
    TRACTOR("Traktör", "wallpapers/nature.jpg", "🚜"),
    TANK("Tank", "wallpapers/nature.jpg", "🛡️"),
    HELICOPTER("Helikopter", "wallpapers/nature.jpg", "🚁"),
    AIRPLANE("Uçak", "wallpapers/space.jpg", "✈️"),
    BICYCLE("Bisiklet", "wallpapers/nature.jpg", "🚲"),
    NATURE("Doğa", "wallpapers/nature.jpg", "🏔️")
}
